### PR TITLE
Rewrite dbparser tests based on criterion

### PR DESCRIPTION
**Pull request by norberttak**
rewrite test_timer_wheel, test_patternize, test_radix unit tests of dbparser based on criterion framework

### CODE REVIEW

**Comment by norberttak** [comment]
@kira-syslogng test this please

File: modules/dbparser/tests/test_patternize.c [hunk 1]
```diff
@@ -334,25 +399,12 @@ testcase_find_clusters_slct(gchar *logs, guint support, gchar *expected)
...
-          fprintf(stderr, "Got clusters:\n");
-          g_hash_table_foreach(clusters, _debug_print2, NULL);
-
-          fail = TRUE;
-        }
+      cr_expect_not(!test_cluster || test_cluster->loglines->len != expected_support,
+                    "expected_cluster='%s', expected_support='%d'\nInput:\n%s\n", expected_item[0], expected_support, param->logs);
 
       g_free(find_data);
-//      g_strfreev(expected_line_strings); // FIXME: this segfaults, so let's leak it instead :)
```
Comment (Kokan) [inline modules/dbparser/tests/test_patternize.c]: Removing this line do not really solve the original problem, I would either won't touch it or try resolving.

Comment (norberttak) [reply modules/dbparser/tests/test_patternize.c]: expected_line_strings variable doesn't exist in updated code. as far as I see any other dynamicaly allocated elemets are cleaned up. let me know if I missunderstand something.

Comment (Kokan) [reply modules/dbparser/tests/test_patternize.c]: Yes, you're right my bad.

File: modules/dbparser/tests/test_radix.c [hunk 2]
```diff
@@ -32,199 +32,112 @@
...
+  else
     {
-      gsize i = 1;
-      gchar *name, *value;
+      cr_assert(ret, "not found while expected: '%s' => none %s\n", key, search_pattern[0]);
 
-      name = name1;
-      value = va_arg(args, gchar *);
-      while (name)
+      for (int i=0; search_pattern[i]; i+=2)
```
Comment (furiel) [inline modules/dbparser/tests/test_radix.c]: For future refactor idea: the name values could be put into a structure, thus avoid the +=2 complexity.

File: modules/dbparser/tests/test_patternize.c [hunk 3]
```diff
@@ -31,18 +31,23 @@
...
+  cfg_load_module(configuration, "syslogformat");
+  msg_format_options_defaults(&parse_options);
+  msg_format_options_init(&parse_options, configuration);
 }
 
-static void _debug_print2(gpointer key, gpointer value, gpointer dummy)
+static void teardown(void)
 {
-  fprintf(stderr, "%s: %d\n", (gchar *) key, ((Cluster *) value)->loglines->len);
+  log_tags_global_deinit();
```
Comment (Kokan) [inline modules/dbparser/tests/test_patternize.c]: Missing deinit calls like `app_shutdown`. (The original test was also missing this.)

Comment (norberttak) [reply modules/dbparser/tests/test_patternize.c]: Missing deinit functions added

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**